internal/contractregistry/contractstore.go

```diff
@@ -58,17 +65,22 @@ type contractStore struct {
 	contractRegistrations map[string]*ContractInfo
 	idxLock               sync.Mutex
 	abiIndex              map[string]messages.TimeSortable
+	abiCache              *lru.Cache
 }
 
-func NewContractStore(baseURL, storagePath string, rr RemoteRegistry) ContractStore {
-	return &contractStore{
+func NewContractStore(baseURL, storagePath string, rr RemoteRegistry) (cs ContractStore, err error) {
+	c := &contractStore{
 		baseURL:               baseURL,
 		storagePath:           storagePath,
 		rr:                    rr,
 		contractIndex:         make(map[string]messages.TimeSortable),
 		contractRegistrations: make(map[string]*ContractInfo),
 		abiIndex:              make(map[string]messages.TimeSortable),
 	}
+	if c.abiCache, err = lru.New(DefaultABICacheSize); err != nil {
```

Comment:
Would it make sense to add a config option for that?

I see the caller has to pass in config explicitly, with `storagePath` being an input. Might be worth having a `ContractStoreConfig` object, similar to other structs.

Not a strong objection here on this one to starting with a `const`, so will leave it with you whether to do this before merge.